libsignal-service/src/provisioning/cipher.rs

```diff
@@ -0,0 +1,223 @@
+use aes::Aes256;
+use block_modes::{block_padding::Pkcs7, BlockMode, Cbc};
+use bytes::Bytes;
+use hmac::{Hmac, Mac, NewMac};
+use prost::Message;
+use rand::Rng;
+use sha2::Sha256;
+
+use libsignal_protocol::{
+    keys::{KeyPair, PublicKey},
+    Context,
+};
+
+pub use crate::proto::{
+    ProvisionEnvelope, ProvisionMessage, ProvisioningVersion,
+};
+
+use crate::{
+    envelope::{CIPHER_KEY_SIZE, IV_LENGTH, IV_OFFSET},
+    provisioning::ProvisioningError,
+};
+
+#[derive(Debug)]
+enum CipherMode {
+    Decrypt(KeyPair),
+    Encrypt(PublicKey),
+}
```

Comment:
Still a bit unclear tbh., but changing this to `DecryptAndEncrypt` and `EncryptOnly` would probably be better. Not part of this PR though, I suppose. (BTW, Rust Analyzer has a great rename function for this).